### PR TITLE
Fix/checkbox react stately

### PR DESCRIPTION
**Pull request by vidhi499**

### CODE REVIEW

File: example/storybook/stories/components/primitives/Checkbox/controlledCheckbox.tsx [hunk 1]
```diff
@@ -2,16 +2,21 @@ import React from 'react';
...
 
 export default function ControlledCheckbox() {
-  const [groupValues, setGroupValues] = React.useState<Array<any>>([]);
+  const [groupValues, setGroupValues] = React.useState<Array<string>>([]);
 
   return (
-    <CheckboxGroup onChange={setGroupValues} value={groupValues}>
-      <Checkbox value="one">
+    <>
+      <Checkbox value="three" isChecked>
```
Comment (intergalacticspacehighway) [inline example/storybook/stories/components/primitives/Checkbox/controlledCheckbox.tsx:9]: @vidhi499 can we remove this isChecked example from here. I think this was just added while testing

File: src/components/primitives/Checkbox/Checkbox.tsx [hunk 2]
```diff
@@ -62,49 +82,61 @@ const Checkbox = ({ icon, ...props }: ICheckboxProps, ref: any) => {
...
+        )}
+      </Center>
+      {props.children}
+    </Box>
+  );
+
+  return (
+    <>
+      {Platform.OS === 'web' ? (
+        <label ref={mergeRefs([ref, ref1])}>
```
Comment (intergalacticspacehighway) [inline src/components/primitives/Checkbox/Checkbox.tsx:126]: Can we assign mergeRefs to a variable?

File: src/components/primitives/Checkbox/CheckboxGroup.tsx [hunk 3]
```diff
@@ -1,42 +1,40 @@
...
-      value={{
-        ...formControlContext,
-        size,
-        colorScheme,
-        onChange,
-        value: values,
-      }}
-    >
-      <Box alignItems="flex-start" {...restCheckboxGroupProps} {...props}>
+    <View {...groupProps}>
```
Comment (intergalacticspacehighway) [inline src/components/primitives/Checkbox/CheckboxGroup.tsx:25]: Can we use the original Box here instead of adding View?

File: package.json [hunk 4]
```diff
@@ -91,7 +92,7 @@
       "<rootDir>/lib/"
     ],
     "transformIgnorePatterns": [
-      "node_modules/(?!(react-native|styled-components|expo-asset|expo-constants|@unimodules|react-native-unimodules|expo-font|react-native-svg|@expo/vector-icons|styled-system|react-native-vector-icons|@react-native-aria/interactions|@react-native-aria/button|@react-native-aria/switch|@react-native-aria/toggle|@react-native-aria/utils|@react-native-aria/radio)/)"
+      "node_modules/(?!(react-native|styled-components|expo-asset|expo-constants|@unimodules|react-native-unimodules|expo-font|react-native-svg|@expo/vector-icons|styled-system|react-native-vector-icons|@react-native-aria/checkbox|@react-native-aria/interactions|@react-native-aria/button|@react-native-aria/switch|@react-native-aria/toggle:|@react-native-aria/utils|@react-native-aria/*)/)"
```
Comment (theankurkedia) [inline package.json:95]: I see `:` after toggle. Is it intentional?

File: src/components/primitives/Checkbox/CheckboxGroup.tsx [hunk 5]
```diff
@@ -1,42 +1,40 @@
...
+import type { ICheckboxGroupProps, ICheckboxContext } from './types';
 import Box from '../Box';
-import type { ICheckboxGroupProps } from './types';
-import { useCheckboxGroup } from './useCheckboxGroup';
 
-export const CheckboxContext = React.createContext({});
+export let CheckboxGroupContext = createContext<ICheckboxContext | null>(null);
 
-const CheckboxGroup = ({
+export function CheckboxGroup({
```
Comment (theankurkedia) [inline src/components/primitives/Checkbox/CheckboxGroup.tsx:13]: ChecboboxGroup is also the default export in this file. Please remove this export

**Pull request merged**